save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/DemoComponent.kt

```diff
@@ -0,0 +1,258 @@
+/**
+ * Function component for different demos
+ */
+
+@file:Suppress("FILE_NAME_MATCH_CLASS")
+
+package com.saveourtool.save.frontend.components.basic
+
+import com.saveourtool.save.demo.diktat.*
+import com.saveourtool.save.frontend.components.basic.codeeditor.codeEditorComponent
+import com.saveourtool.save.frontend.externals.fontawesome.*
+import com.saveourtool.save.frontend.externals.reactace.AceModes
+import com.saveourtool.save.frontend.externals.reactace.AceThemes
+import com.saveourtool.save.frontend.utils.*
+
+import csstype.ClassName
+import org.w3c.dom.asList
+import org.w3c.files.FileReader
+import react.*
+import react.dom.aria.AriaRole
+import react.dom.aria.ariaLabel
+import react.dom.html.ButtonType
+import react.dom.html.InputType
+import react.dom.html.ReactHTML.button
+import react.dom.html.ReactHTML.div
+import react.dom.html.ReactHTML.h4
+import react.dom.html.ReactHTML.hr
+import react.dom.html.ReactHTML.input
+import react.dom.html.ReactHTML.label
+import react.dom.html.ReactHTML.p
+import react.dom.html.ReactHTML.strong
+
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+
+val diktatDemoComponent = diktatDemoComponent()
+
+private val diktatDemoDefaultCode = """
```

Comment:
suggestion to move an example on backend side -- it can be different for different demo tool

Reply:
I see no reason why we might want to move it to the backend as far as we do not fetch test data from backend - only result file and warnings

Reply:
we might want/have to to fetch it from backend for demo tool which doesn't support kotlin